hive/src/main/java/org/apache/iceberg/hive/legacy/LegacyHiveTableOperations.java

```diff
@@ -109,13 +109,14 @@ protected void doRefresh() {
     if (current().spec().fields().isEmpty()) {
       matchingDirectories = ImmutableList.of(getDirectoryInfo());
     } else {
-      matchingDirectories = getDirectoryInfosByFilter(expression);
+      matchingDirectories = getDirectoryInfosByFilter(expression, current().spec());
     }
 
-    Iterable<Iterable<DataFile>> filesPerDirectory = Iterables.transform(matchingDirectories, directory -> {
```

Comment:
this was getting hard for me to read, hence had to format these methods a little. Here and other places